apps/designer/app/shared/array-utils.ts

```diff
@@ -1,11 +1,23 @@
-export const filterMutable = <T>(
-  array: T[],
-  predicate: (value: T) => boolean
-) => {
+type Predicate<T> = (value: T) => boolean;
```

Comment:
Please don't use abbreviation, I know its what ts documentation uses, but its a know bad idea that ts authors expressed several times

Reply:
could be Item in this case e.g.